naga/src/front/wgsl/parse/number.rs

```diff
@@ -286,11 +296,14 @@ fn parse(input: &str) -> (Result<Number, NumberError>, &str) {
 
                 let digits_with_sign = general_extract.end(bytes);
 
+                // Don't accept the `l` suffix here; a decimal point
+                // or exponent is required for `f64` literals. We
+                // might want to use `l` for 64-bit integers as well.
```

Comment:
I think we should keep things consistent with `f`/`h` where it's not required to have a decimal point or exponent if the suffix has been specified.

The only way to do that is to disambiguate the syntax by having the suffixes map to a single type.

I like rust's suffixes but those are not consistent with the WGSL spec.

I suggest using `lf` for `f64`.
Then, we can have `li` and `lu` for `i64`/`u64`.

We can also use the same `SK` (`S` = size, `K` = kind) scheme later for 8-bit and 16-bit types (with the only exception being `hf` = `h`).